packages/ui/react-ui-editor/src/components/TextEditor/TextEditor.tsx

```diff
@@ -39,61 +29,43 @@ export type CursorInfo = {
   after?: string;
 };
 
-export type TextEditorSlots = {
-  root?: Omit<ComponentProps<'div'>, 'ref'>;
-  editor?: {
-    className?: string;
-  };
-  content?: {
-    className?: string;
-  };
-};
-
 export type TextEditorProps = {
-  model: EditorModel; // TODO(burdon): Optional (e.g., just provide content if readonly).
-  readonly?: boolean; // TODO(burdon): Move into model.
+  dataTestId?: string; // TODO(burdon): Generalize?
+  doc?: string; // TODO(burdon): Rename text/value/content?
+  className?: string;
   autoFocus?: boolean;
-  scrollPastEnd?: boolean;
-  moveToEndOfLine?: boolean;
-  lineWrapping?: boolean;
   scrollTo?: StateEffect<any>; // TODO(burdon): Restore scroll position: scrollTo EditorView.scrollSnapshot().
+  moveToEndOfLine?: boolean;
   selection?: { anchor: number; head?: number };
-  placeholder?: string;
-  theme?: ThemeStyles;
-  slots?: TextEditorSlots;
   extensions?: Extension[];
   debug?: boolean;
 };
 
 /**
- * Base text editor.
+ * Thin wrapper for text editor.
+ * Handles tabster and focus management.
  */
-// TODO(burdon): Replace with useTextEditor.
-export const BaseTextEditor = forwardRef<EditorView | null, TextEditorProps>(
+export const TextEditor = forwardRef<EditorView | null, TextEditorProps>(
   (
     {
-      model,
-      readonly,
+      doc,
+      className,
       autoFocus,
       scrollTo = EditorView.scrollIntoView(0),
       moveToEndOfLine,
       selection,
-      theme,
-      slots = defaultSlots,
       extensions = [],
-      scrollPastEnd: _scrollPastEnd,
       debug,
     },
     forwardedRef,
   ) => {
+    // TODO(burdon): Make tabster optional.
     const tabsterDOMAttribute = useFocusableGroup({ tabBehavior: 'limited' });
```

Comment:
tabster could be moved inside `useTextEditor` and it could return dom attributes to provide to the parent alongside the ref

Reply:
+1